src/components/SidebarChatList.tsx

```diff
@@ -1,16 +1,20 @@
 "use client";
 
-import React, {FC, useState} from "react";
+import React, {FC, useEffect, useState} from "react";
 import SidebarChatListItem from "@/components/SidebarChatListItem";
 interface SidebarChatListProps {
     friends: User[],
     sessionId: string,
 }
 
 const SidebarChatList:FC<SidebarChatListProps> = ({friends, sessionId})=>{
-    const [activeChats] = useState<User[]>(friends);
+    const [activeChats, setActiveChats] = useState<User[]>(friends);
     const [unseenMessages] = useState<Message[]>([]);
 
+    useEffect(() => {
```

Comment:
This is different from the demo, but if this solution gives us the needed update behavior without sending more messaging, it might be a good tradeoff.